pkgs/by-name/pi/pint/package.nix

```diff
@@ -0,0 +1,71 @@
+{ lib
+, buildGoModule
+, fetchFromGitHub
+, makeWrapper
+, curl
+, perl
+, git
+, nix-update-script
+, testers
+, pint
+}:
+
+buildGoModule rec {
+  pname = "pint";
+  version = "0.51.0";
+
+  src = fetchFromGitHub {
+    owner = "cloudflare";
+    repo = "pint";
+    rev = "v${version}";
+    hash = "sha256-d3KZPPeQJBqdrr81YLusYHc5jLChC1Rf5SYeP/QMeo8=";
+  };
+
+  vendorHash = "sha256-9tJL33Qtu0xAmRnAP0BwM6CIfGs+GEG864e89XrpfLM=";
+
+  nativeBuildInputs = [ makeWrapper ];
+
+  subPackages = [ "cmd/pint" ];
+
+  CGO_ENABLED = 0;
+
+  ldflags = [
+    "-s"
+    "-w"
+    "-X=main.version=${version}"
+    "-X=main.commit=${src.rev}"
+  ];
+
+  preCheck = ''
+    substituteInPlace internal/git/git.go \
+      --replace '.Command("git"' '.Command("${lib.getExe git}"'
```

Comment:
Now you are patching the file _and_ wrapping the program.

Reply:
I know, I think it's best to do it that way, the program is compiled first, so the source code is not modified. Then the tests are ran, so the code is only patched to make the tests pass.

Reply:
I too support only wrapping the program, and not patching it like that, as the substitutions could become unnoticed warnings in the future, and they are a maintenance overhead. As for the tests, you can use `nativeCheckInputs` for the packages needed for testing.